specific/cmo_sale/models/sale.py

```diff
@@ -67,39 +68,66 @@ class SaleOrder(models.Model):
         states={'done': [('readonly', True)]},
         default=lambda self: self._get_th_convenant(),
     )
+    discount_type = fields.Selection(
+        [('percent', 'Percentage'),
+         ('amount', 'Amount')],
+         inverse='_compute_discount_rate',
+         string='Discount Type',
+         states={'done': [('readonly', True)]},
+    )
+    discount_rate = fields.Float(
+        string='Discount Rate',
+        inverse='_compute_discount_rate',
+        store=True,
+        states={'done': [('readonly', True)]},
+    )
+    amount_discount = fields.Float(
+        string="Discount",
+        compute='_compute_amount_discount',
+        readonly=True,
+    )
+
+    @api.depends('amount_untaxed')
+    def _compute_amount_discount(self):
+        total = sum(self.order_line.mapped('price_unit'))
+        self.amount_discount = total - self.amount_untaxed
 
     @api.multi
+    @api.depends('amount_before_management_fee')
     def _compute_before_management_fee(self):
-        total = 0
-        for line in self.order_line:
-            if line.order_lines_group == 'before':
-                total = total + line.price_subtotal
+        total = sum(self.order_line.filtered(\
```

Comment:
Thanks for trying to use filtered.
If you use @api.multi, you should use the following too. Otherwise it is not correct for multiple sales order cases.
for rec in self: